ConsoleAppAzureServices/ConsoleAppAzureServices/Program.cs

```diff
@@ -1,47 +1,49 @@
-﻿using System;
+﻿using Azure;
+using Azure.Identity;
+using Azure.ResourceManager.Resources;
+using Azure.ResourceManager.Resources.Models;
+using Azure.Security.KeyVault.Secrets;
+using System;
 using System.Threading.Tasks;
-using Microsoft.Azure.KeyVault;
-using Microsoft.Azure.Management.ResourceManager;
-using Microsoft.Azure.Services.AppAuthentication;
-using Microsoft.Rest;
 
 namespace ConsoleAppAzureServices
 {
     class Program
     {
         static void Main()
         {
-            AzureServiceTokenProvider azureServiceTokenProvider = new AzureServiceTokenProvider();
+            InteractiveBrowserCredential credential = new InteractiveBrowserCredential();
 
-            GetSecretFromKeyVault(azureServiceTokenProvider).Wait();
+            AuthenticationRecord authRecord = credential.Authenticate();
 
-            GetResourceGroups(azureServiceTokenProvider).Wait();
+            GetSecretFromKeyVault(credential).Wait();
 
-            if (azureServiceTokenProvider.PrincipalUsed != null)
+            GetResourceGroups(credential).Wait();
+
+            if (credential != null)
             {
-                Console.WriteLine($"{Environment.NewLine}Principal used: {azureServiceTokenProvider.PrincipalUsed}");
+                Console.WriteLine($"{Environment.NewLine}Principal used:{authRecord.Authority} TenantId:{authRecord.TenantId} UserPrincipalName:{authRecord.Username}");
             }
 
             Console.ReadLine();
         }
 
-        private static async Task GetSecretFromKeyVault(AzureServiceTokenProvider azureServiceTokenProvider)
+        private static async Task GetSecretFromKeyVault(InteractiveBrowserCredential credential)
```

Comment:
You should not be mixing sync `Main` and async code like this. If you're going to be async (recommended), use `async Task Main` and `await` all async calls. If not and using a sync `Main`, just make this method sync as well and call sync methods on `SecretClient`.